File: C#/Logic/LinearSearch3.cs
using C_.Class;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_.Logic
{
    internal class LinearSearch3 : LogicSearch
    {
        /// <summary>
        /// 線形探索法（break版）
        /// </summary>
        public override void Search()
        {
            Person[] person = new Person[]
            {
                new Person("Ann", 18),
                new Person("Rolla", 19),
                new Person("Nancy", 16),
                new Person("Eluza", 17),
                new Person("Juliet", 18),
                new Person("Candy", 16)
            };

            string keyName = "Candy";
            bool searchFlg = false;

            for (int i = 0; i < person.Length; i++)
            {
                if (person[i].Name.Equals(keyName))
                {
                    Console.WriteLine(person[i].Name + " " + person[i].Age);
                    searchFlg = true;
                }
            }

            if (!searchFlg)
            {
                Console.WriteLine("見つかりませんでした。");
            }
        }
    }
}

File: Java/Search/src/Logic/LinearSearch3.java
package Logic;

import Class.Person;

/**
 * 線形探索法（break版）
 */
public class LinearSearch3 extends LogicSearch {
	public void Search() {
		Person[] p = new Person[6];
		p[0] = new Person("Ann", 18);
		p[1] = new Person("Rolla", 19);
		p[2] = new Person("Nancy", 16);
		p[3] = new Person("Eluza", 17);
		p[4] = new Person("Juliet", 18);
		p[5] = new Person("Machilda", 20);
		
		int dataCount = p.length;
		String keyName = "Juliet";
		boolean flg = false;
		
		for(int i = 0; i < dataCount; i++) {
			if(keyName == p[i].getName()) {
				System.out.println(p[i].getName() + " " + p[i].getAge());
				flg = true;
				break;
			}
		}
		
		if(!flg) {
			System.out.println("見つかりませんでした");
		}
	}
}
